source/scripts/storage/storage.js

```diff
@@ -1,23 +1,58 @@
-//storage {
-//   categories: {cat1:[names],cat2:[names]}
-//   savedLists: {list1:[names],list2:[names]}
-//   recipeName1: json object
-//   recipeName2: json object
+// localStorage = {
+//   categories: { // Used to populate explore page (only ids to reduce storage size)
+//       "breakfast": [
+//           "123", // recipe id
+//           "111", // recipe id
+//           "444", // recipe id
+//       ],
+//       "lunch": [
+//           "123", // recipe id
+//           "111", // recipe id
+//           "444", // recipe id
+//       ],
+//   },
+//   savedRecipes: { // Saved recipe page categories (only ids to reduce storage size)
+//       "favorites": [
+//           "123", // recipe id
+//           "111", // recipe id
+//           "444", // recipe id
+//       ],
+//       "list2": [
+//           "123", // recipe id
+//           "111", // recipe id
+//           "444", // recipe id
+//       ],
+//   },
+//   recipeData: { // Complete storage of available recipes we can query without fetching
+//       "111": "full recipe json object", // these are api results so we never have to modify them
+//       "123": "full recipe json object",
+//       "444": "full recipe json object",
+//       "555": "full recipe json object",
+//       "715": "full recipe json object",
+//       "892": "full recipe json object"
+//   }
 // }
 
+
 // store recipe to local storage
-// add recipe name to category
+// add recipeid to category
 // input: category of api fetch, array of recipe json object
 export function storeRecipeData(category, recipeArray) {
   try {
+    let allData = JSON.parse(localStorage.getItem("recipeData"));
+    if (allData == null) {
+      allData = {};
+    }
+    let currNumElement = Object.keys(allData).length;
+
     // store recipe
+    let allRecipeId = [];
     for (let i = 0; i < recipeArray.length; i++) {
-      localStorage.setItem(
-        recipeArray[i]["title"],
-        JSON.stringify(recipeArray[i])
-      );
+      const uid = currNumElement + i;
```

Comment:
the spoonacular api gives us a unique id for each recipe when we fetch our initial search results.
```
{
    "id": 716429,
    "title": "Pasta with Garlic, Scallions, Cauliflower & Breadcrumbs",
    "calories": 584,
    "carbs": "84g",
    "fat": "20g",
    "image": "https://spoonacular.com/recipeImages/716429-312x231.jpg",
    "imageType": "jpg",
    "protein": "19g"
},
```
 we have to use that id to fetch the full recipe data later. that's kinda what i meant by using the id as the key but this might work fine too im not sure

Reply:
Oh lol ! I see. Let's use this id then.